convert2rhel/logger.py

```diff
@@ -88,6 +90,35 @@ def initialize_logger(log_name, log_dir=LOG_DIR):
     logger.addHandler(handler)
 
 
+def archive_old_logger_files():
+    """Archive the old log files to not mess with multiple runs outputs.
+    Every time a new run begins, this method will be called to archive the previous logs
+    if there is a `convert2rhel.log` file there, it will be archived using
+    the same name for the log file, but having an appended timestamp to it.
+
+    For example:
+        /var/log/convert2rhel/archive/convert2rhel-1635162445070567607.log
+        /var/log/convert2rhel/archive/convert2rhel-1635162478219820043.log
+
+    This way, the user can track the logs for each run individually based on the timestamp.
+    """
+
+    current_log_file = "%s/convert2rhel.log" % LOG_DIR
```

Comment:
Looks like we're hardcoding convert2rhel.log here but `initialize_logger()` allows it to be set by the caller.  So this needs to be set by the caller as well. If you like my proposed change: https://github.com/oamg/convert2rhel/pull/359/files#r741280923 it's natural to keep those in sync as the function that's the entrypoint will be able to pass the value to both helper functions.

Reply:
You're right! It's indeed way better to just keep them in sync and avoid duplication of code! I just wonder about this line https://github.com/oamg/convert2rhel/pull/359/files/6beb9a275de4891162a42a7dc2cbe9d5020c180c#diff-f82efabf7bb126100afd0472e3d8a5902da6576c6ddf75e3184cce9eb647c52aR118, since it will be used by another tool to collect the archived logs, I wonder If we can reuse the `log_name` to create the archived log file? I mean, they already accepted the PR on `main` branch https://github.com/sosreport/sos/pull/2734, but I can always open another one to fix the path if we need some changes. Do you have any thoughts on that one?

Reply:
Yep, I agree that we should use log_name to both lookup the current log file and as part of the name of the archived log.

I don't think we need another change to sosreport right now.  We're hardcoding log_name as convert2rhel.log when we call initialize_logger() in main.py here: https://github.com/oamg/convert2rhel/pull/359/files/6beb9a275de4891162a42a7dc2cbe9d5020c180c#diff-7d31913eebe96d308744cc192ecf12ed870f425d573064d6e5dad3b53381ecc4R62

If we allow user's to customize the log name in the future, sosreport will need to be changed for both the current log name and the archived log name.